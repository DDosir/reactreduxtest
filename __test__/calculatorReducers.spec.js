import React from 'react'
import { shallow, mount } from 'enzyme';

import calculatorReducers from '../src/js/reducers/calculatorReducers'

describe('>>>R E D U C E R --- Test calculatorReducers',()=>{
    it('+++ reducer for ADD_INPUT', () => {
        let state = {output:100}
        state = calculatorReducers(state,{type:"ADD_INPUTS",output:500})
        expect(state).toEqual({output:500})
    });
    it('+++ reducer for SUBTRACT_INPUT', () => {
        let state = {output:100}
        state = calculatorReducers(state,{type:"SUBTRACT_INPUTS",output:50})
        expect(state).toEqual({output:50})
    });

    it('+++ reducer with shapshot', () => {
      expect(calculatorReducers(undefined, { type: 'default' })).toMatchSnapshot();
    });

    it('+++ reducer with shapshot', () => {
      const action = {
        type: 'ADD_INPUTS',
        output: 60,
      };
      expect(calculatorReducers(undefined, action)).toMatchSnapshot();
    });

});
//*******************************************************************************************************
