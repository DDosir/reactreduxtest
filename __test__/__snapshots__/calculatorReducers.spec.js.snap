// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>R E D U C E R --- Test calculatorReducers +++ reducer with shapshot 1`] = `
Object {
  "output": 0,
}
`;

exports[`>>>R E D U C E R --- Test calculatorReducers +++ reducer with shapshot 2`] = `
Object {
  "output": 50,
}
`;
